mysql-test/suite/period/r/alter.result

```diff
@@ -0,0 +1,49 @@
+set @s= '1992-01-01';
+set @e= '1999-12-31';
+create or replace table t (s date, e date);
+alter table t add period for a(s, e);
+ERROR HY000: Period field `s` cannot be NULL
+alter table t change s s date, add period for a(s, e);
+ERROR HY000: Period field `e` cannot be NULL
+# NOT NULL is added implicitly if field is altered
+alter table t change s s date, change e e date, add period for a(s, e);
```

Comment:
This is questionable. Is this written in standard? To me (and to simple user as well) the command
```sql
alter table t change s s date, change e e date, add period for a(s, e);
```
looks equivalent to
```sql
alter table t add period for a(s, e);
```
because CHANGE COLUMN doesn't change anything.

Reply:
SQL standard allows implicit NOT NULL. Unfortunately, CHANGE COLUMN is not standard clause. But I am trying to be consistent here:
1. CHANGE COLUMN behavior shouldn't differ from ADD COLUMN
2. ADD COLUMN works same as if it was declared in CREATE TABLE
3. implicit NOT NULL is allowed
4. Columns that are not altered explicitly are left unchanged

Reply:
According to spec:
```
h) BC1 and BC2 shall be known not nullable.
```
It's better to make 1.

Reply:
I guess it should be read as:
> shall be known not *as* nullable

BC1 and BC2 are columns of current table schema:
```
e) ... The table descriptor of T shall include a column descriptor that includes BCN1 as the column name. The column BC1 identified by BCN1 is the ATPN period start column.
```
The standard does not support several ADD clauses (except some cases like `ADD SYSTEM_TIME ADD column ADD column`):
```
11.10 <alter table statement>

<alter table statement> ::=
ALTER TABLE <table name> <alter table action>
```

Reply:
At last, the decision has been made to disallow implicit NOT NULL for both ADD COLUMN and CHANGE COLUMN